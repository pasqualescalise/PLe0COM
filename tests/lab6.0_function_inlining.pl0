PROCEDURE function_to_inline_no_return();
	BEGIN
		print 10 + 20
	END;

PROCEDURE function_to_inline_no_return_string();
	BEGIN
		print "Hello World!"
	END;

PROCEDURE function_to_inline_return() -> (int);
	BEGIN
		return (10 + 20)
	END;

PROCEDURE function_not_to_inline();
	VAR result1, result2 : int;
	BEGIN
		CALL function_to_inline_no_return();
		CALL function_to_inline_no_return_string();
		CALL function_to_inline_no_return();
		CALL function_to_inline_return() -> (result1);
		print result1
	END;

BEGIN
	CALL function_not_to_inline()
END.
