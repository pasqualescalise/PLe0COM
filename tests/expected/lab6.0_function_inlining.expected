30
Hello World!
30
30
30
20
30
30
200
300
199
199
